api/include/opentelemetry/trace/trace_state.h

```diff
@@ -45,51 +45,149 @@ class TraceState
-
-    // Creates an Entry for a given key-value pair.
-    Entry(nostd::string_view key, nostd::string_view value) noexcept;
-
-    nostd::string_view GetKey();
-    nostd::string_view GetValue();
+    Entry() : key_(nullptr), value_(nullptr){};
+
+    // Copy constructor
+    Entry(Entry &copy)
+    {
+      key_   = StringToPointer(nostd::string_view(copy.key_.get()));
+      value_ = StringToPointer(nostd::string_view(copy.value_.get()));
+    }
+
+    // Assignment operator
+    Entry &operator=(Entry &other)
+    {
+      key_   = StringToPointer(nostd::string_view(other.key_.get()));
+      value_ = StringToPointer(nostd::string_view(other.value_.get()));
+      return *this;
+    }
+
+    // Create an Entry for a given key-value pair.
+    Entry(nostd::string_view key, nostd::string_view value) noexcept
+    {
+      key_   = StringToPointer(key);
+      value_ = StringToPointer(value);
+    }
+
+    // Get the key associated with this entry.
+    nostd::string_view GetKey() { return nostd::string_view(key_.get()); }
+
+    // Get the value associated with this entry.
+    nostd::string_view GetValue() { return nostd::string_view(value_.get()); }
+
+    // Set the value for this entry. This overrides the previous value.
+    void SetValue(nostd::string_view value) { value_ = StringToPointer(value); }
 
   private:
     // Store key and value as raw char pointers to avoid using std::string.
     nostd::unique_ptr<const char[]> key_;
     nostd::unique_ptr<const char[]> value_;
+
+    // Copy string into a buffer and return a unique_ptr to the buffer.
+    // This is a workaround for the fact that strcpy doesn't accept a const char* destination.
+    nostd::unique_ptr<const char[]> StringToPointer(nostd::string_view str)
+    {
+      nostd::unique_ptr<char[]> temp(new char[str.size() + 1]);
+      strcpy(temp.get(), str.data());
+      return nostd::unique_ptr<const char[]>(temp.release());
+    }
   };
 
   // An empty TraceState.
-  TraceState() noexcept : num_entries_(0) {}
+  TraceState() noexcept : entries_(new Entry[kMaxKeyValuePairs]), num_entries_(0) {}
 
-  // Returns false if no such key, otherwise returns true and populates value.
-  bool Get(nostd::string_view key, nostd::string_view value) const noexcept { return false; }
+  // Returns false if no such key, otherwise returns true and populates the value parameter with the
+  // associated value.
+  bool Get(nostd::string_view key, nostd::string_view &value) noexcept
+  {
+    for (int i = 0; i < num_entries_; i++)
+    {
+      if (key == (entries_.get())[i].GetKey())
+      {
+        value = (entries_.get())[i].GetValue();
+        return true;
+      }
+    }
+    return false;
+  }
 
   // Creates an Entry for the key-value pair and adds it to entries.
   // If value is null, this function is a no-op.
-  void Set(nostd::string_view key, nostd::string_view value) const noexcept;
+  void Set(nostd::string_view key, nostd::string_view value) noexcept
```

Comment:
Can we return a boolean here, so the user can now if the value was added successfully?